src/topology/topological_fiber_bundle.lean

```diff
@@ -146,125 +155,201 @@ open_locale topological_space classical
+    simp only [mem_preimage, e.proj_symm_apply hx.2],
+    exact hx.1, },
+  { rw [←inter_univ univ, ←prod_inter_prod, mem_inter_eq] at hx,
+    exact hx.2, }
+end
 
-lemma bundle_trivialization.mk_proj_snd' (e : bundle_trivialization F proj) {x : Z}
-  (ex : proj x ∈ e.base_set) : (proj x, (e x).2) = e x :=
+end prebundle_trivialization
+
+variable [topological_space Z]
+
+/--
+A structure extending local homeomorphisms, defining a local trivialization of a projection
+`proj : Z → B` with fiber `F`, as a local homeomorphism between `Z` and `B × F` defined between two
+sets of the form `proj ⁻¹' base_set` and `base_set × F`, acting trivially on the first coordinate.
+-/
+@[nolint has_inhabited_instance]
+structure bundle_trivialization (proj : Z → B) extends local_homeomorph Z (B × F) :=
+(base_set      : set B)
+(open_base_set : is_open base_set)
+(source_eq     : source = proj ⁻¹' base_set)
+(target_eq     : target = set.prod base_set univ)
+(proj_to_fun   : ∀ p ∈ source, (to_local_homeomorph p).1 = proj p)
+
+namespace bundle_trivialization
+
+variables {F} (e : bundle_trivialization F proj) {x : Z}
+
+/-- Natural identification as `prebundle_trivialization`. -/
+def to_prebundle_trivialization : prebundle_trivialization F proj := { ..e }
+
+instance : has_coe_to_fun (bundle_trivialization F proj) := ⟨_, λ e, e.to_fun⟩
+instance : has_coe (bundle_trivialization F proj) (prebundle_trivialization F proj) :=
+⟨to_prebundle_trivialization⟩
+
+@[simp, mfld_simps] lemma coe_coe : ⇑e.to_local_homeomorph = e := rfl
+@[simp, mfld_simps] lemma coe_fst (ex : x ∈ e.source) : (e x).1 = proj x := e.proj_to_fun x ex
+protected lemma eq_on : eq_on (prod.fst ∘ e) proj e.source := λ x hx, e.coe_fst hx
+lemma mem_source : x ∈ e.source ↔ proj x ∈ e.base_set := by rw [e.source_eq, mem_preimage]
+lemma coe_fst' (ex : proj x ∈ e.base_set) : (e x).1 = proj x := e.coe_fst (e.mem_source.2 ex)
+lemma mk_proj_snd (ex : x ∈ e.source) : (proj x, (e x).2) = e x := prod.ext (e.coe_fst ex).symm rfl
+lemma mk_proj_snd' (ex : proj x ∈ e.base_set) : (proj x, (e x).2) = e x :=
 prod.ext (e.coe_fst' ex).symm rfl
 
-protected lemma bundle_trivialization.eq_on (e : bundle_trivialization F proj) :
-  eq_on (prod.fst ∘ e) proj e.source :=
-λ x hx, e.coe_fst hx
+@[simp, mfld_simps] lemma coe_mk (e : local_homeomorph Z (B × F)) (i j k l m) (x : Z) :
+  (bundle_trivialization.mk e i j k l m : bundle_trivialization F proj) x = e x := rfl
+
+lemma mem_target {x : B × F} : x ∈ e.target ↔ x.1 ∈ e.base_set :=
+by rw [e.target_eq, prod_univ, mem_preimage]
+
+lemma map_target {x : B × F} (hx : x ∈ e.target) : e.to_local_homeomorph.symm x ∈ e.source :=
+e.to_local_homeomorph.map_target hx
 
-lemma bundle_trivialization.proj_symm_apply (e : bundle_trivialization F proj) {x : B × F}
-  (hx : x ∈ e.target) : proj (e.to_local_homeomorph.symm x) = x.1 :=
+lemma proj_symm_apply {x : B × F} (hx : x ∈ e.target) : proj (e.to_local_homeomorph.symm x) = x.1 :=
```

Comment:
No it does't... lean complains that the `e` in `hx` is a prebundle trivialization and not a bundle one... This would be different if the structures were extending one another? Do you have solutions to propose? I also don't like to rewrite a proof that needs more than two lines!

Reply:
That's really weird: I just tried it on my computer, and it works fine. I mean
```lean
lemma proj_symm_apply {x : B × F} (hx : x ∈ e.target) : proj (e.to_local_homeomorph.symm x) = x.1 :=
e.to_prebundle_trivialization.proj_symm_apply hx
```
just works. Have you tried it?

Reply:
Ops I was trying
```
prebundle_trivialization.proj_symm_apply ↑e hx
```
that I was was completely equivalent... Why does it change?